cni-plugin/kubernetes.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// newKubeClient is a unit test override variable for interface create.
+var newKubeClient = newK8sClient
+
+// getKubePodInfo is a unit test override variable for interface create.
+var getKubePodInfo = getK8sPodInfo
+
+// newK8sClient returns a Kubernetes client
+func newK8sClient(conf PluginConf, logger *logrus.Entry) (*kubernetes.Clientset, error) {
```

Comment:
This is ok for a POC, but we already have a lot of code in the project for configuring a Kubernetes client and reading pod info from the Kubernetes API. Most of it lives in `pkg/k8s` and `controller/k8s`. For consistency my preference would be to use that instead.

Reply:
Now using pkg/k8s